File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManagerInterface.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.uimanager;

import android.view.View;
import androidx.annotation.Nullable;
import com.facebook.react.bridge.Dynamic;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReadableMap;

/**
 * This is an interface that should be implemented by view managers supporting the base view
 * properties such as backgroundColor, opacity, etc.
 */
public interface BaseViewManagerInterface<T extends View> {
  void setAccessibilityActions(T view, @Nullable ReadableArray accessibilityActions);

  void setAccessibilityHint(T view, @Nullable String accessibilityHint);

  void setAccessibilityLabel(T view, @Nullable String accessibilityLabel);

  void setAccessibilityLiveRegion(T view, @Nullable String liveRegion);

  void setAccessibilityRole(T view, @Nullable String accessibilityRole);

  void setAccessibilityCollection(T view, @Nullable ReadableMap accessibilityCollection);

  void setAccessibilityCollectionItem(T view, @Nullable ReadableMap accessibilityCollectionItem);

  void setViewState(T view, @Nullable ReadableMap accessibilityState);

  void setBackgroundColor(T view, int backgroundColor);

  void setBorderRadius(T view, float borderRadius);

  void setBorderBottomLeftRadius(T view, float borderRadius);

  void setBorderBottomRightRadius(T view, float borderRadius);

  void setBorderTopLeftRadius(T view, float borderRadius);

  void setBorderTopRightRadius(T view, float borderRadius);

  void setElevation(T view, float elevation);

  void setFilter(T view, ReadableArray filter);

  void setShadowColor(T view, int shadowColor);

  void setImportantForAccessibility(T view, @Nullable String importantForAccessibility);

  void setRole(T view, @Nullable String role);

  void setNativeId(T view, @Nullable String nativeId);

  void setAccessibilityLabelledBy(T view, @Nullable Dynamic nativeId);

  void setOpacity(T view, float opacity);

  void setRenderToHardwareTexture(T view, boolean useHWTexture);

  void setRotation(T view, float rotation);

  void setScaleX(T view, float scaleX);

  void setScaleY(T view, float scaleY);

  void setTestId(T view, String testId);

  void setTransform(T view, @Nullable ReadableArray matrix);

  void setTransformOrigin(T view, @Nullable ReadableArray transformOrigin);

  void setTranslateX(T view, float translateX);

  void setTranslateY(T view, float translateY);

  void setZIndex(T view, float zIndex);
}

File: packages/react-native/temp/package/ReactAndroid/src/main/java/com/facebook/react/uimanager/BaseViewManagerInterface.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.uimanager

import android.view.View
import com.facebook.react.bridge.Dynamic
import com.facebook.react.bridge.ReadableArray
import com.facebook.react.bridge.ReadableMap

/**
 * This is an interface that should be implemented by view managers supporting the base view
 * properties such as backgroundColor, opacity, etc.
 */
public interface BaseViewManagerInterface<T : View> {
  public fun setAccessibilityActions(view: T, accessibilityActions: ReadableArray?)

  public fun setAccessibilityHint(view: T, accessibilityHint: String?)

  public fun setAccessibilityLabel(view: T, accessibilityLabel: String?)

  public fun setAccessibilityLiveRegion(view: T, liveRegion: String?)

  public fun setAccessibilityRole(view: T, accessibilityRole: String?)

  public fun setAccessibilityCollection(view: T, accessibilityCollection: ReadableMap?)

  public fun setAccessibilityCollectionItem(view: T, accessibilityCollectionItem: ReadableMap?)

  public fun setViewState(view: T, accessibilityState: ReadableMap?)

  public fun setBackgroundColor(view: T, backgroundColor: Int)

  public fun setBorderRadius(view: T, borderRadius: Float)

  public fun setBorderBottomLeftRadius(view: T, borderRadius: Float)

  public fun setBorderBottomRightRadius(view: T, borderRadius: Float)

  public fun setBorderTopLeftRadius(view: T, borderRadius: Float)

  public fun setBorderTopRightRadius(view: T, borderRadius: Float)

  public fun setElevation(view: T, elevation: Float)

  public fun setFilter(view: T, filter: ReadableArray)

  public fun setMixBlendMode(view: T, setMixBlendMode: String)

  public fun setShadowColor(view: T, shadowColor: Int)

  public fun setImportantForAccessibility(view: T, importantForAccessibility: String?)

  public fun setRole(view: T, role: String?)

  public fun setNativeId(view: T, nativeId: String?)

  public fun setAccessibilityLabelledBy(view: T, nativeId: Dynamic?)

  public fun setOpacity(view: T, opacity: Float)

  public fun setRenderToHardwareTexture(view: T, useHWTexture: Boolean)

  public fun setRotation(view: T, rotation: Float)

  public fun setScaleX(view: T, scaleX: Float)

  public fun setScaleY(view: T, scaleY: Float)

  public fun setTestId(view: T, testId: String?)

  public fun setTransform(view: T, matrix: ReadableArray?)

  public fun setTransformOrigin(view: T, transformOrigin: ReadableArray?)

  public fun setTranslateX(view: T, translateX: Float)

  public fun setTranslateY(view: T, translateY: Float)

  public fun setZIndex(view: T, zIndex: Float)
}
